scrape/manager.go

```diff
@@ -318,6 +325,7 @@ func (m *Manager) ApplyConfig(cfg *config.Config) error {
 					sp.logger.Error("No logger found. This is a bug in Prometheus that should be reported upstream.", "scrape_pool", name)
 				}
 			}
+			<-canReload
```

Comment:
Should this not also be part of a cleanup via defer?

Reply:
moving it to defer would just be a different way of calling it, but doesn't change anything, does it?